sklearn/preprocessing/_encoders.py

```diff
@@ -526,8 +526,8 @@ def _validate_keywords(self):
         # in interpreting the model.
         if self.drop is not None and self.handle_unknown != 'error':
             raise ValueError(
-                "`handle_unknown` cannot be 'ignore' when the drop parameter "
-                "is specified, as this will create ambiguous cells")
+                "`handle_unknown` must be 'error' when the drop parameter "
+                "is specified to avoid ambiguities in inverse_transform")
```

Comment:
It's not just about `inverse_transform`. `transform` would map both unkown and dropped categories to all zeros so the transformed output would be ambiguous.
May I suggest

```
`handle_unknown` must be 'error' when the drop parameter is
specified, as both would create categories that are all-zeros.
```